doc/development/powershell.md

```diff
@@ -0,0 +1,165 @@
+This page contains guidelines for developing or updating powershell scripts used by the Azure SDK engineering system.
+
+Table of Contents
+=================
+
+* [TLDR](#tldr)
+* [Structure](#structure)
+* [Functionality](#functionality)
+* [Style](#style)
+* [Testing](#testing)
+   * [Unit Testing](#unit-testing)
+      * [Running Pester Tests](#running-pester-tests)
+   * [Local/Pipeline Functional Testing](#localpipeline-functional-testing)
+
+## TLDR
+
+_(i.e. too long, didn't read)_
+
+**DO**
+
+- Try to write scripts that can be run locally
+- Write unit tests for your scripts
+- Handle exit codes from external programs
+- Use `Set-StrictMode -Version 4` where possible
+- Define your data structures as much as possible via composition, or unit test or comment examples.
+- Use an editor extension or `Invoke-ScriptAnalyzer` to analyze code.
```

Comment:
If you're going to use a "DO" headline, I'd be more assertive here e.g., drop the "try", etc. Alternatively, put them in a "SHOULD" headline.

Reply:
Yeah that's a good point. Clearly I'm not sure how assertive I want to be :P